odoo/addons/base/tests/test_expression.py

```diff
@@ -1773,12 +2055,14 @@ def test_autojoin(self):
         with self.assertQueries(['''
             SELECT "res_partner"."id"
             FROM "res_partner"
-            LEFT JOIN "res_country" AS "res_partner__country_id" ON
-                ("res_partner"."country_id" = "res_partner__country_id"."id")
             LEFT JOIN "res_company" AS "res_partner__company_id" ON
                 ("res_partner"."company_id" = "res_partner__company_id"."id")
-            WHERE (("res_partner__company_id"."name" LIKE %s)
-                OR ("res_partner__country_id"."code" LIKE %s))
+            LEFT JOIN "res_country" AS "res_partner__country_id" ON
+                ("res_partner"."country_id" = "res_partner__country_id"."id")
+            WHERE (
+                "res_partner__company_id"."name" LIKE %s
+                OR "res_partner__country_id"."code" LIKE %s
+            )
```

Comment:
The reordering of LEFT JOINs is a good thing in this case, since it follows the needs of the conditions :+1:   The former order reflected the implementation which was parsing the domain right-to-left.